src/Http/Routing/src/DataSourceDependentCache.cs

```diff
@@ -8,6 +8,8 @@
 
 namespace Microsoft.AspNetCore.Routing
 {
+    // FYI: This class is also linked into MVC. If you make changes to the API you will
+    // also need to change MVC's usage.
     internal sealed class DataSourceDependentCache<T> : IDisposable where T : class
```

Comment:
This commit has a bunch of random cleanup because I'm trying to share the core data structure used in action selection in a few places.